pkg/autoscaler/autoscaler.go

```diff
@@ -344,6 +344,17 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 					workloadArch = arch
 					expectedScaledMachineSet = machineSet
 				}
+
+				It("should have expected upstream annotations", func() {
+					//checking for the newly added upstream annotations from the CAO
+					expectedAnnotations := map[string]string{
+						"machine.openshift.io/vCPU":     "",
+						"machine.openshift.io/memoryMb": "",
+						"machine.openshift.io/GPU":      "",
+					}
+
+					Expect(machineSet.Annotations).To(HaveValue(Equal(expectedAnnotations)))
```

Comment:
ultimately we want a different test here.

we probably want to use `HaveKey`, https://pkg.go.dev/github.com/onsi/gomega#HaveKey to test that the keys we want are present.

then for the values we want to do a light validation, so for the CPU and memory values (on the `capacity.cluster.x-k8s.io` annotations) we want to ensure that a `resource.Quantity.MustParse`, https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#MustParse, will succeed. for the GPU count we want to make sure that it is parseable as a number, and for GPU type that it is a string with length greater than 0.

hope this makes sense, but if not you know where to find me =)